core/src/main/java/com/linecorp/armeria/server/DefaultUnhandledExceptionsReporter.java

```diff
@@ -63,35 +53,19 @@ final class ExceptionReportingServerErrorHandler implements ServerErrorHandler,
     @Nullable
     private Throwable thrownException;
 
-    ExceptionReportingServerErrorHandler(MeterRegistry meterRegistry, ServerErrorHandler serverErrorHandler,
-                                         Duration interval) {
-        assert !interval.isNegative() && !interval.isZero() : interval;
-        this.interval = interval;
-        delegate = serverErrorHandler;
+    DefaultUnhandledExceptionsReporter(MeterRegistry meterRegistry, EventLoopGroup workerGroup,
+                                       long intervalMillis) {
+        this.workerGroup = workerGroup;
+        this.intervalMillis = intervalMillis;
         micrometerCounter = meterRegistry.counter("armeria.server.exceptions.unhandled");
         counter = new LongAdder();
     }
 
-    /**
-     * Increments both {@code micrometerCounter} and {@code counter} and stores unhandled exception
-     * that occurs. If multiple servers use the same {@link MeterRegistry}, then {@code micrometerCounter}
-     * will be shared across servers.
-     */
-    @Nullable
     @Override
-    public HttpResponse onServiceException(ServiceRequestContext ctx, Throwable cause) {
-        if (!ctx.shouldReportUnhandledExceptions()) {
-            return delegate.onServiceException(ctx, cause);
-        }
-
-        if ((cause instanceof HttpStatusException || cause instanceof HttpResponseException) &&
-            cause.getCause() != null) {
-            return delegate.onServiceException(ctx, cause);
-        }
-
+    public void report(Throwable cause) {
         if (reportingTaskFuture == null && scheduledUpdater.compareAndSet(this, 0, 1)) {
-            reportingTaskFuture = ctx.eventLoop().scheduleAtFixedRate(
-                    this::reportException, interval.toMillis(), interval.toMillis(), TimeUnit.MILLISECONDS);
+            reportingTaskFuture = workerGroup.next().scheduleAtFixedRate(
```

Comment:
nit: Directly use `CommonPools.workerGroup()`? Without testing purposes, we wouldn't take advantage of the injected `workerGroup`.